modules/http-client/src/main/scala/pillars/httpclient/httpclient.scala

```diff
@@ -27,12 +42,80 @@ final case class HttpClient[F[_]: Async](client: org.http4s.client.Client[F])
     extends pillars.Module[F]:
     export client.*
 
+    def callSecure[SI, I, EO, O, R](
+        endpoint: Endpoint[SI, I, EO, O, R],
+        uri: Option[Uri],
+        clientOptions: Http4sClientOptions = Http4sClientOptions.default
+    )(securityInput: SI, input: I): F[Either[EO, O]] =
+        callRequest(endpoint, uri)(
+          Http4sClientInterpreter[F](clientOptions).toSecureRequest(endpoint, uri)(securityInput)(input)
+        )
```

Comment:
this will allocate an interpreter per request. I suggest using a closure as return type `(SI, I) => F[Either[EO, O]]` and close over the interpreter.